### PR TITLE
Increment sequence number on transaction failure

### LINKED ISSUE #67: Increment the proposer key sequence for failed transactions
### Problem

The `flow-emulator` doesn't increment the proposer key sequence for failed transactions. The current behavior doesn't match the changes introduced in [#602](https://github.com/onflow/flow-go/pull/602) which was released in flow [v0.17.4](https://github.com/onflow/flow-go/releases/tag/v0.17.4). I only tested the two cases listed below but most likely there are more cases to cover.

Version: `v.0.22.0`

### Steps to Reproduce

1. Submit a transfer transaction with an amount larger than the sender balance. The transaction should fail and the key sequence should be incremented. 
```
➜  flow transactions get fcfbfae19116a4c6c30646a430533900c.....

❌ Transaction Error
execution error code 1101: [Error Code: 1101] cadence runtime error Execution failed:
error: pre-condition failed: Amount withdrawn must be less than or equal than the balance of the Vault
   --> ee82856bf20e2aa6.FungibleToken:170:16
    |
170 |                 self.balance >= amount:
    |                 ^^^^^^^^^^^^^^^^^^^^^^
Status		✅ SEALED
ID		fcfbfae19116a4c6c30646a430533900c...
Payer		8913a7d76722a585
Authorizers	[8913a7d76722a585]
Proposal Key:
    Address	8913a7d76722a585
    Index	0
    Sequence	1
```
The key sequence remains 1 after the transaction is sealed.
```
➜  flow accounts get 8913a7d76722a585
Address	 0x8913a7d76722a585
Balance	 1.00100000
Keys	 1
Key 0	Public Key		 bcee61165014006481673cef8dbb204daf5f33e369....
	Weight			 1000
	Signature Algorithm	 ECDSA_secp256k1
	Hash Algorithm		 SHA3_256
	Revoked 		 false
	Sequence Number 	 1
	Index 			 0
```

2. Submit a transfer transaction to an invalid address or an address that cannot receive the transfer token "not setup". The transaction should fail and the key sequence should be incremented. 
```
➜  flow transactions get 1c06fd999d93c57313056dfae458...
❌ Transaction Error
execution error code 1101: [Error Code: 1101] cadence runtime error Execution failed:
error: panic: Could not borrow receiver reference to the recipient's Vault
  -->1c06fd999d93c57313056dfae458...:27:6
   |
27 | 			?? panic("Could not borrow receiver reference to the recipient's Vault")
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Status		✅ SEALED
ID		1c06fd999d93c57313056dfae458...
Payer		700a49a02359d112
Authorizers	[700a49a02359d112]
Proposal Key:
    Address	700a49a02359d112
    Index	0
    Sequence	1
```
The sequence remains 1 after the transaction is exuted
```
➜  flow accounts get 700a49a02359d112
Address	 0x700a49a02359d112
Balance	 1.00100000
Keys	 1
Key 0	Public Key		 349c2dab77ebf344f971b5a23fc44...
	Weight			 1000
	Signature Algorithm	 ECDSA_secp256k1
	Hash Algorithm		 SHA3_256
	Revoked 		 false
	Sequence Number 	 1
	Index 			 0
```

### Acceptance Criteria

Match the flow implementation by incrementing the proposer key sequence for all transactions, except the four cases mentioned in this [announcement](https://forum.onflow.org/t/proposer-key-sequence-number-breaking-change/1913).

>- invalid transaction (transaction rejected by collection nodes)
>- any of the transaction signatures are not valid
>- the transaction is missing the signature(s) by the proposer
>- the proposal sequence number is invalid (not match with the one stored on the chain)

### Context

N/A

### PR DESCRIPTION
**Pull request by sideninja**
Closes #67 

## Description
The sequence number on the proposal key wasn't increased in case of failed transactions since the view wasn't merged and stored. This PR introduces a test for that case and fixes the problem.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: blocks.go [hunk 1]
```diff
@@ -21,16 +21,12 @@ package emulator
...
 
-	"github.com/onflow/flow-go/access"
 	"github.com/onflow/flow-go/fvm"
 	flowgo "github.com/onflow/flow-go/model/flow"
 
 	"github.com/onflow/flow-emulator/storage"
 )
 
-var _ fvm.Blocks = &blocks{}
-var _ access.Blocks = &blocks{}
```
Comment (turbolent) [inline blocks.go:32]: Why remove the type assertions?

File: pendingBlock.go [hunk 2]
```diff
@@ -165,13 +165,12 @@ func (b *pendingBlock) ExecuteNextTransaction(
...
-		if err != nil {
-			// fail fast if fatal error occurs
-			return nil, err
-		}
+	b.events = append(b.events, tp.Events...)
+
+	err = b.ledgerView.MergeView(childView)
+	if err != nil {
+		// fail fast if fatal error occurs
+		return nil, err
```
Comment (turbolent) [inline pendingBlock.go:173]: @ramtinms Could you please have a look and check this matches the real behaviour in flow-go?

**Review by turbolent — COMMENTED** [review]
Nice

File: blocks.go (hunk 1 above)
Comment (sideninja) [reply blocks.go:32]: Ohh, I thought this was meant to silent an unused warning :/ sorry not yet familiar with some patterns in Go. Will revert.

**Review by psiemens — APPROVED** [review]
Looks good to me, but I agree with @turbolent that it'd be good to get @ramtinms's eyes on this!

File: pendingBlock.go (hunk 2 above)
Comment (janezpodhostnik) [reply pendingBlock.go:173]: The change is good IMO. A transaction error should not prevent merging the state or events.

But maybe @ramtinms can take a peek as well

**Pull request merged**